common/source/docs/common-blheli32-passthru.rst

```diff
@@ -1,62 +1,158 @@
+Reversible DShot (aka 3D mode) allows the motor to be spun in either direction which is important for Rover, Boats and :ref:`Planes with reverse thrust <plane:reverse-thrust-setup>`.
+
+Currently, only BLHeli32 and BLHeli_S capable reversible DShot ESCs are supported. In order to use one, the output which drives it must be designated with the appropriate bit in the :ref:`SERVO_BLH_3DMASK<SERVO_BLH_3DMASK>` bitmask parameter. This will map the outputs 1000-1500-2000 values to the correct digital values for the ESC to provide FullReverse-Idle-FullForward range operation, respectively.
+
+If :ref:`DShot commands <dshot-commands>` have been enabled then ArduPilot will automatically configure the ESCs to reversible mode (3D mode) at startup, according to the :ref:`SERVO_BLH_3DMASK<SERVO_BLH_3DMASK>`. Enabling :ref:`DShot commands <dshot-commands>` will allow the other DShot commands to be sent to any other ESC configured as DShot by the DShot mask parameters discussed in :ref:`DShot setup instructions <common-dshot-escs>`.
 
-If the craft has been setup for DShot commands then ArduPilot will supply the correct command at startup in order to set the ESCs in reversible mode.
+Otherwise, you must manually configure the ESCs' "Motor Direction" to "Bidirectional 3D" as shown below.
 
-In a similar fashion, normal output rotation direction can be reversed by setting :ref:`SERVO_BLH_RVMASK<SERVO_BLH_RVMASK>` without any changes needing to be made through ESC setup software (e.g. BLHeliSuite). This can also be used on ESCs with forward and reversed active operation, ie reversible ESCs, to set the "forward" direction's rotation.
+  .. image:: ../../../images/blheli-reversible-dshot.png
+    :target: ../_images/blheli-reversible-dshot.png
+    :width: 450px
 
 .. note:: Currently, ArduPilot only supports the use of reversible ESCs for Plane and Rover, not Copter.
 
-BLHeli32 ESC Telemetry
-----------------------
+.. _blheli32-esc-telemetry:
+
+ESC Telemetry
+-------------
+
+If the ESC has this capability, it allows monitoring and logging of performance data that previously required additional sensors (like power modules and RPM sensors). The detailed data provided by every ESC allows real-time decisions and individual ESC or motor performance tuning and failure analysis. Note that a given ESC may or may not have a specific sensor's data transmitted via telemetry. It is common for 4 in 1 escs to provide voltage and current sensors but not transmit the data via telemetry, but rather by direct connection to the autopilot. Check the ESC data sheet and connection information for details.
```

Comment:
Should note that ESC telemetry is specific to dshot - it will not work with PWM/OneShot etc even if the ESC has the signal wire

Reply:
Txs.  I don't think we actually need to mention PWM or OneShot because right at the top of this page it says, "Before attempting to use BLHeli please follow the DShot setup instructions".  I think it could actually muddy the waters to mention them.. we just have to focus on concise and precise instructions to get it working.. if users start doing things that are not described .. well.. that's hard to protect against.